src/test/java/nl/dtls/fairdatapoint/service/impl/PurlPIDSystemImplTest.java

```diff
@@ -88,45 +90,66 @@ public void testGetURIForNullMetadataUri() {
         metadataCopy.setUri(null);
         test.getURI(metadataCopy);
     }
-    
-    
+
     /**
      * Test of null fdp uri, this test is excepted to throw error
      */
     @Test(expected = NullPointerException.class)
     public void testGetURIForNullFdpUri() {
-        PurlPIDSystemImpl pisdSys =  new PurlPIDSystemImpl();
-        pisdSys.getURI(catalogMetadata);
+        test.getURI(catalogMetadata);
+    }
+
+    /**
+     * Test of null parent uri, this test is excepted to throw error
+     */
+    @Test(expected = NullPointerException.class)
+    public void testGetURIForNullParentUri() {
+        CatalogMetadata mdata = catalogMetadata;
+        mdata.setParentURI(null);
+        test.getURI(mdata);
     }
-    
-    
+
     /**
      * Test of valid fdpMetadata uri, this test is excepted to pass
+     *
+     * @throws java.lang.NoSuchFieldException If the filed is not declared
      */
     @Test
-    public void testGetURIForValidMetadata() {
-        assertTrue(test.getURI(fdpMetadata).toString().contains("purl.org"));
-        assertTrue(test.getURI(catalogMetadata).toString().contains("purl.org"));
-        assertTrue(test.getURI(datasetMetadata).toString().contains("purl.org"));
-        assertTrue(test.getURI(distributionMetadata).toString().contains("purl.org"));
+    public void testGetURIForValidMetadata() throws NoSuchFieldException {
+
+        PurlPIDSystemImpl mock = Mockito.mock(PurlPIDSystemImpl.class);
+        // Setting up mock object
+        when(mock.getURI(fdpMetadata)).thenReturn(valueFactory
+                .createIRI("http://purl.org/biosemantics-lumc/fdp"));
+        when(mock.getURI(catalogMetadata)).thenReturn(valueFactory
+                .createIRI("http://purl.org/biosemantics-lumc/fdp/catalog/1"));
+        when(mock.getURI(datasetMetadata)).thenReturn(valueFactory
+                .createIRI("http://purl.org/biosemantics-lumc/fdp/dataset/1"));
+        when(mock.getURI(distributionMetadata)).thenReturn(valueFactory
+                .createIRI("http://purl.org/biosemantics-lumc/fdp/distribution/1"));
+
+        assertTrue(mock.getURI(fdpMetadata).toString().contains("purl.org"));
+        assertTrue(mock.getURI(catalogMetadata).toString().contains("purl.org"));
+        assertTrue(mock.getURI(datasetMetadata).toString().contains("purl.org"));
+        assertTrue(mock.getURI(distributionMetadata).toString().contains("purl.org"));
```

Comment:
This test is not testing our code at all, this is testing mockito. For proper testing of the system under test, we need to mock and inject the underlying dependencies of `PurlPIDSystemImpl`, but leave the instance itself a real class instance.